File: 04-Matrizes/Java/Dado.java
public class Dado {

    int valor;
    int linha;
    int coluna;

    public Dado(int valor, int linha, int coluna) {
        this.valor = valor;
        this.linha = linha;
        this.coluna = coluna;
    }

    public int getValor() {
        return valor;
    }

    public void setValor(int valor) {
        this.valor = valor;
    }

    public int getLinha() {
        return linha;
    }

    public void setLinha(int linha) {
        this.linha = linha;
    }

    public int getColuna() {
        return coluna;
    }

    public void setColuna(int coluna) {
        this.coluna = coluna;
    }

   

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        
        final Dado other = (Dado) obj;

        if (this.linha != other.linha) {
            return false;
        }
        return this.coluna == other.coluna;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Dado{");
        sb.append("valor=").append(valor);
        sb.append(", linha=").append(linha);
        sb.append(", coluna=").append(coluna);
        sb.append('}');
        return sb.toString();
    }





}

File: 04-Matrizes/Python/Dado.py
class Dado:
    def __init__(self, valor, linha, coluna) :
        self.valor = valor
        self.linha = linha
        self.coluna = coluna

    def getValor(self):
        return self.valor
    
def __eq__(self, other):
    if isinstance(other, Dado):
        return self.linha == other.linha and self.coluna == other.coluna
    return False

def __str__(self):
    return f"Dado(valor={self.valor}, linha={self.linha}, coluna={self.coluna})"